rust/src/ctypes.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use ffi_utils::strings::{
+    string_to_c_char,
+    c_char_to_string,
+};
+use std::os::raw::c_char;
+use std::ptr;
+
+use mentat_core::Uuid;
+use time::Timespec;
+
+use items::{
+    Item, Items
+};
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+pub struct ItemC {
+    pub uuid: *mut c_char,
+    pub name: *mut c_char,
+    pub due_date: *mut i64,
```

Comment:
nit: half of these fields are using an alias, and half aren't. maybe either use i8 instead of c_char, or libc's time_t instead of i64?

Reply:
And we refer to c_chars in helper methods, so maybe `time_t` then? (super nit)